<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-07-04" />

<title>2024-07-04_SimulateIndirectlyComparingTreatments</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2024_ComparativeSplicing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">2024-07-04_SimulateIndirectlyComparingTreatments</h1>
<h4 class="date">2024-07-04</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#intro" id="toc-intro">Intro</a></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-07-09
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>2024_comparativesplicing/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed19900924code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(19900924)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed19900924code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(19900924)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong5495e15">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 5495e15
</a>
</p>
</div>
<div id="strongRepositoryversionstrong5495e15"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 5495e15.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.snakemake/
    Ignored:    code/GenomeFiles/
    Ignored:    code/SJC_annotations_d5d1043c36aa60f014dee37e6f54418aeac524c63858131c0c68769dba597410.pckle
    Ignored:    code/Session.vim
    Ignored:    code/kaessmanAnalysis/
    Ignored:    code/kaessman_AS_dat/
    Ignored:    code/logs/
    Ignored:    code/rules/.GenomeAnnotations.smk.swp
    Ignored:    code/scratch/

Untracked files:
    Untracked:  analysis/2024-07-08_CheckLeafcutter2_Classifications.Rmd
    Untracked:  code/scripts/LeafcutterJuncClassificationsToAS_segments.R

Unstaged changes:
    Modified:   code/Snakefile
    Modified:   code/envs/bedparse.yml
    Modified:   code/rules/GenomeAnnotations.smk
    Modified:   code/scripts/AS_ExonSegments_to_AS_Junctions.py
    Modified:   code/scripts/leafcutter2

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/2024-07-04_SimulateIndirectlyComparingTreatments.Rmd</code>)
and HTML
(<code>docs/2024-07-04_SimulateIndirectlyComparingTreatments.html</code>)
files. If you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
5495e15
</td>
<td>
Benjmain Fair
</td>
<td>
2024-07-08
</td>
<td>
updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
5495e15
</td>
<td>
Benjmain Fair
</td>
<td>
2024-07-08
</td>
<td>
updates
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>Simple model for expression change after treatment for a single
gene</p>
<p><span class="math inline">\(Y = Y_0 + \beta + \epsilon\)</span></p>
<p><span class="math inline">\(Y\)</span> is measured expression level
of a gene after treatment, <span class="math inline">\(Y_0\)</span> is
expression of control, <span class="math inline">\(\beta\)</span> is
effect size, <span class="math inline">\(\epsilon\)</span> is some
random error (<span class="math inline">\(\mathcal{N}(\mu=0,
\sigma)\)</span>). Similarly, the measured expression of a control
sample is</p>
<p><span class="math inline">\(Y = Y_0 + \epsilon\)</span></p>
<p>One could expand this to vector notation for multiple samples, but
why bother for this toy example.</p>
<p>And of course, real DE analysis is more complicated than these simple
models because of how it borrows information across genes to estimate
dispersion per gene… But let’s keep this simple.</p>
<p>Now, consider 10000 genes. I’m going to simulate some stuff, saving
the variable names similar to notation above but keeping in mind each is
a vector now representing <span class="math inline">\(Y_0\)</span>,
<span class="math inline">\(Y\)</span>, etc, across 10000 genes…</p>
<pre class="r"><code># &quot;baseline&quot; (control) expression level.
# Let&#39;s say true control expression across genes is lognormal, or rather normal on a log scale
# let&#39;s use some parameters that somewhat resemble real RNA-seq data for log2RPKM across 1000 genes.
Y_0 &lt;- rnorm(mean = 4, sd = 3, n = 10000)
hist(Y_0)</code></pre>
<p><img src="figure/2024-07-04_SimulateIndirectlyComparingTreatments.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
5495e15
</td>
<td>
Benjmain Fair
</td>
<td>
2024-07-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Now simulate some betas. Let’s say the betas (representing log2FC)
are also normally distributed. And let’s make two different vectors of
betas, for two different (and completely independent treatments with
uncorrellated) true effects. I’m trying to pick parameters for these
normal distributions that sort of match my intuitinos on what would be
really measured in an RNA-seq experiment, where like ~90% of genes have
less than two-fold change. But different treatments have different beta
vectors (across genes), and my intuitions might be off on what the
distribution of “true” effects is. We can play with these parameters
later.</p>
<pre class="r"><code>beta_treatment1 &lt;- rnorm(mean=0, sd=0.5, n=10000)
hist(beta_treatment1)</code></pre>
<p><img src="figure/2024-07-04_SimulateIndirectlyComparingTreatments.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>beta_treatment2 &lt;- rnorm(mean=0, sd=0.5, n=10000)
hist(beta_treatment2)</code></pre>
<p><img src="figure/2024-07-04_SimulateIndirectlyComparingTreatments.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(beta_treatment1, beta_treatment2)</code></pre>
<p><img src="figure/2024-07-04_SimulateIndirectlyComparingTreatments.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># betas are uncorrelated
correlation &lt;- cor.test(beta_treatment1, beta_treatment2)
print(correlation$estimate)</code></pre>
<pre><code>        cor 
0.002531988 </code></pre>
<p>betas for two treatments are uncorrelated (R=0.002532).</p>
<p>Ok, now let’s simulate some real observed values by adding the random
error term. Let’s simulate two control samples, and one of each
treatment. How does the random measurement error compare to the true
effect sizes. Well I suppose that depends on the nature of the treatment
(eg, the amount of drug, the amount of hypoxia, etc) among other
things.</p>
<pre class="r"><code># Let&#39;s use a slightly smaller sigma for the error term for now
error_sigma &lt;- 0.5

observed.control.1 &lt;- Y_0 + rnorm(mean=0, sd=error_sigma, n=10000)
observed.control.2 &lt;- Y_0 + rnorm(mean=0, sd=error_sigma, n=10000)

observed.treatment.1 &lt;- Y_0 + beta_treatment1 +  rnorm(mean=0, sd=error_sigma, n=10000)

observed.treatment.2 &lt;- Y_0 + beta_treatment2 +  rnorm(mean=0, sd=error_sigma, n=10000)</code></pre>
<p>Now let’s compare the differences in observed expression between
different samples.</p>
<pre class="r"><code>plot(observed.control.1, observed.control.2)</code></pre>
<p><img src="figure/2024-07-04_SimulateIndirectlyComparingTreatments.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>print(cor(observed.control.1, observed.control.2))</code></pre>
<pre><code>[1] 0.9724647</code></pre>
<p>Just like real RNA-seq, control replicates correlate very well
(R&gt;0.95)</p>
<p>Now let’s plot <span class="math inline">\(Y-Y_0\)</span>, an
estimate of treatment effects. Let’s estimate effects both treatments,
using independent control samples for each… start with treatment 1…</p>
<pre class="r"><code># estimate of betas (across genes)
beta_treatment1_estimate &lt;- observed.treatment.1 - observed.control.1
hist(beta_treatment1_estimate)</code></pre>
<p><img src="figure/2024-07-04_SimulateIndirectlyComparingTreatments.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># true betas
hist(beta_treatment1)</code></pre>
<p><img src="figure/2024-07-04_SimulateIndirectlyComparingTreatments.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># scatter plot
plot(beta_treatment1, beta_treatment1_estimate)</code></pre>
<p><img src="figure/2024-07-04_SimulateIndirectlyComparingTreatments.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>…And same for treatment2</p>
<pre class="r"><code># estimate of betas (across genes)
beta_treatment2_estimate &lt;- observed.treatment.2 - observed.control.2
hist(beta_treatment2_estimate)</code></pre>
<p><img src="figure/2024-07-04_SimulateIndirectlyComparingTreatments.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># true betas
hist(beta_treatment2)</code></pre>
<p><img src="figure/2024-07-04_SimulateIndirectlyComparingTreatments.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># scatter plot
qplot(beta_treatment2, beta_treatment2_estimate)</code></pre>
<p><img src="figure/2024-07-04_SimulateIndirectlyComparingTreatments.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>And confirm that beta estimates for two treatments are uncorrelated
across genes…</p>
<pre class="r"><code>qplot(beta_treatment1_estimate, beta_treatment2_estimate) +
  geom_smooth(method=&#39;lm&#39;) +
  geom_hline(yintercept = 0)</code></pre>
<pre><code>`geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="figure/2024-07-04_SimulateIndirectlyComparingTreatments.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>print(cor.test(beta_treatment1_estimate, beta_treatment2_estimate))</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  beta_treatment1_estimate and beta_treatment2_estimate
t = 2.972, df = 9998, p-value = 0.002966
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.01011564 0.04928120
sample estimates:
       cor 
0.02970982 </code></pre>
<p>Ok, now let’s repeat this the less kosher way, that reuses controls…
In this case I will just average the controls. Not really what is done,
but I think this simpler example still highlights the issue. While it
makes sense to combine controls for maximum power if you are just
concerned with estimating betas for one treatment, when comparing the
beta estimates between treatments, this can lead to some misleading
results.</p>
<pre class="r"><code>beta_treatment1_estimate2 &lt;- observed.treatment.1 - (observed.control.1 + observed.control.2)/2

beta_treatment2_estimate2 &lt;- observed.treatment.2 - (observed.control.1 + observed.control.2)/2

qplot(beta_treatment1_estimate2, beta_treatment2_estimate2) +
  geom_smooth(method=&#39;lm&#39;) +
  geom_hline(yintercept = 0)</code></pre>
<pre><code>`geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="figure/2024-07-04_SimulateIndirectlyComparingTreatments.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cor.test(beta_treatment1_estimate2, beta_treatment2_estimate2)</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  beta_treatment1_estimate2 and beta_treatment2_estimate2
t = 22.524, df = 9998, p-value &lt; 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.2010238 0.2383315
sample estimates:
     cor 
0.219758 </code></pre>
<p>Ok, that introduced some artificial correlation. Not sure how
realistic these simulations are though. Maybe the signal to noise of the
treatments in real life is such that it doesn’t matter. Especially with
more than just the couple replicates I simulated here. Or maybe not. And
maybe downstream anaysis like GSEA aren’t sensitive enough to pick up on
this small artificial correlation. Or maybe it is. Using independent
controls before comparing betas between treatments leaves less room for
question. Depending on the question, directly comparing treatments might
also be applicable.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.9     purrr_0.3.4    
[5] readr_2.1.2     tidyr_1.2.0     tibble_3.1.7    ggplot2_3.3.6  
[9] tidyverse_1.3.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.12      lattice_0.20-45  lubridate_1.8.0  assertthat_0.2.1
 [5] rprojroot_2.0.3  digest_0.6.29    utf8_1.2.2       R6_2.5.1        
 [9] cellranger_1.1.0 backports_1.4.1  reprex_2.0.1     evaluate_0.15   
[13] highr_0.9        httr_1.4.3       pillar_1.7.0     rlang_1.0.2     
[17] readxl_1.4.0     rstudioapi_0.13  whisker_0.4      jquerylib_0.1.4 
[21] Matrix_1.5-3     rmarkdown_2.14   splines_4.2.0    labeling_0.4.2  
[25] munsell_0.5.0    broom_0.8.0      compiler_4.2.0   httpuv_1.6.5    
[29] modelr_0.1.8     xfun_0.30        pkgconfig_2.0.3  mgcv_1.8-40     
[33] htmltools_0.5.2  tidyselect_1.1.2 workflowr_1.7.0  fansi_1.0.3     
[37] crayon_1.5.1     tzdb_0.3.0       dbplyr_2.1.1     withr_2.5.0     
[41] later_1.3.0      grid_4.2.0       nlme_3.1-157     jsonlite_1.8.0  
[45] gtable_0.3.0     lifecycle_1.0.1  DBI_1.1.2        git2r_0.30.1    
[49] magrittr_2.0.3   scales_1.3.0     cli_3.6.2        stringi_1.7.6   
[53] farver_2.1.0     fs_1.5.2         promises_1.2.0.1 xml2_1.3.3      
[57] bslib_0.3.1      ellipsis_0.3.2   generics_0.1.2   vctrs_0.4.1     
[61] tools_4.2.0      glue_1.6.2       hms_1.1.1        fastmap_1.1.0   
[65] yaml_2.3.5       colorspace_2.0-3 rvest_1.0.2      knitr_1.39      
[69] haven_2.5.0      sass_0.4.1      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
