<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-09-30" />

<title>2024-09-30_ContinousTimeSplicingModel</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2024_ComparativeSplicing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">2024-09-30_ContinousTimeSplicingModel</h1>
<h4 class="date">2024-09-30</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#intro" id="toc-intro">Intro</a></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>2024_comparativesplicing/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed19900924code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(19900924)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed19900924code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(19900924)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkun2024comparativesplicingtree172f74b99c3d796bd47fa2cac0f2aa13fcd77647targetblank172f74ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/2024_comparativesplicing/tree/172f74b99c3d796bd47fa2cac0f2aa13fcd77647" target="_blank">172f74b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkun2024comparativesplicingtree172f74b99c3d796bd47fa2cac0f2aa13fcd77647targetblank172f74ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/2024_comparativesplicing/tree/172f74b99c3d796bd47fa2cac0f2aa13fcd77647" target="_blank">172f74b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/.Rhistory
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/ChainFiles/
    Ignored:    code/CordosoMoreira_Fastq/
    Ignored:    code/Downloads/
    Ignored:    code/GenomeFiles/
    Ignored:    code/LiftoverJuncs/
    Ignored:    code/Log.out
    Ignored:    code/MazinLeafcutterAnalysis/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/config/OldConfigs/2040822_Cordoso_Moreira_SampleList.tsv
    Ignored:    code/conservation/
    Ignored:    code/featureCounts/
    Ignored:    code/kaessmanAnalysis/
    Ignored:    code/kaessman_AS_dat/
    Ignored:    code/logs/
    Ignored:    code/rna-seq/
    Ignored:    code/scratch/
    Ignored:    code/scripts/.SpearmanCor_Mazin_log2RPKM.R.swp
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/.vscode/
    Ignored:    code/snakemake.log
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/2024-08-21_SpearmanFromPSI_WithinSpecies.Rmd
    Untracked:  analysis/2024-08-21_SpearmanFromPSI_WithinSpecies_AllSpecies.Rmd
    Untracked:  analysis/2024-08-21_SpearmanFromPSI_WithinSpecies_SampleStagesFixed.Rmd
    Untracked:  analysis/2024-08-24_BioMartLookupGenes.Rmd
    Untracked:  analysis/2024-08-29_ExploreJuncLiftovers.Rmd
    Untracked:  analysis/2024-08-29_OrganizeCrossSpeciesTables.Rmd
    Untracked:  analysis/2024-09-06_UnrpdoctuviveSplicingAndExpression.Rmd
    Untracked:  analysis/2024-09-09_Organize_ConserveddevASJuncs.Rmd
    Untracked:  analysis/2024-09-10_RedoMazinJuncAnalysisFixedClassifications.Rmd
    Untracked:  analysis/2024-09-23_FixMouseEnsemblBed12ToGtf.Rmd
    Untracked:  analysis/2024-09-30_ContinousExpressionSplicingModel.Rmd
    Untracked:  analysis/2024-09-30_ContinousTimeSplicingModel.Rmd
    Untracked:  analysis/20240815_LiftoverJuncsTest.ipynb
    Untracked:  analysis/Untitled.ipynb
    Untracked:  code/config/CordosoGenomes_Extra_Gtfs.tsv
    Untracked:  code/config/GTEx_juncFileList.tsv
    Untracked:  code/envs/crossmap.yml
    Untracked:  code/envs/py27.yml
    Untracked:  code/rules/MazinLeafcutterAnalysis.smk
    Untracked:  code/scripts/FeatureCounts_to_Mat.R
    Untracked:  code/scripts/PrepAllJuncsFor_JunctionClassifier.R
    Untracked:  code/scripts/QQNorm_PSITable_ByTissue.R
    Untracked:  code/scripts/SpearmanCor_Mazin_LeafcutterPSI.R
    Untracked:  code/scripts/SpearmanCor_Mazin_log2RPKM.R
    Untracked:  code/scripts/Untitled.ipynb
    Untracked:  code/scripts/leafcutter_to_PSI_GTEX.R
    Untracked:  data/Stages_AsIn_CordosoMoreira.tsv
    Untracked:  data/Stages_AsIn_CordosoMoreira_Recoded.txt
    Untracked:  output/Conserved.devAS.leafcutter.tsv.gz
    Untracked:  output/Ensembl.GeneHumanHomologs.tsv.gz
    Untracked:  output/Ensembl.TranscriptInfo.tsv.gz
    Untracked:  output/GTEx_DS_DE_FromChao_CordosoTissuePairs.tsv.gz

Unstaged changes:
    Modified:   analysis/2024-07-16_Download_CordosoMoreira_Fastq.Rmd
    Modified:   analysis/index.Rmd
    Modified:   code/Snakefile
    Modified:   code/config/ChainFiles.tsv
    Modified:   code/config/Cordoso_Moreira_SampleList.tsv
    Modified:   code/config/STAR_Genome_List.tsv
    Modified:   code/config/samples.tsv
    Modified:   code/envs/bedparse.yml
    Modified:   code/module_workflows/snakemake-workflow_rna-seq
    Modified:   code/rules/LiftoverJuncs.smk
    Modified:   code/rules/common.smk
    Modified:   code/scripts/daiuc_leafcutter2
    Modified:   code/scripts/leafcutter2
    Modified:   output/QC/ReadCountsPerSamples.tsv

Staged changes:
    Modified:   .gitmodules
    New:        code/scripts/daiuc_leafcutter2

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>Had some wierd results using spearman correlation coef as proxy for
effect size (beta) when making the splicing beta vs expression beta
scatter plots across developmental time in various tissues/species.
Mabye that spearman correlation coef is just not the way to go with
leafcutter quantifications. Maybe if we prepare the expression
quantifications a little more carefully (eg standardization and
normalization as in sQTL splicing table preperation), then get ’beta’s
from linear model (Splicing ~ OrdinalDevelopmentalTime) within each
species:tissue combination, the problem will go away. Let’s try setting
up that, here (and possibly later incorporating that work into the
snakemake).</p>
<pre class="r"><code>library(tidyverse)
library(data.table)
library(magrittr)

expression.table &lt;- read_tsv(&quot;../code/MazinLeafcutterAnalysis/Expression/Rabbit_UCSC.oryCun2_ensv101.log2rpkm.tsv.gz&quot;) %&gt;%
  mutate(Geneid = str_replace(Geneid, &quot;^(.+?)\\..+$&quot;, &quot;\\1&quot;))

CordosoSamples &lt;- read_tsv(&quot;../code/config/Cordoso_Moreira_SampleList.tsv&quot;)

expression.table.tidy &lt;- expression.table %&gt;%
  pivot_longer(names_to = &quot;ID&quot;, values_to = &quot;log2expression&quot;, -Geneid) %&gt;%
  inner_join(
    CordosoSamples %&gt;%
      filter(`Used library?` %in% c(&quot;yes&quot;, &quot;Yes&quot;)) %&gt;%
      dplyr::select(ID:ID_Stage, Ordinal_stage, Tissue_ForDevelopementalAnalysis) %&gt;%
      separate_rows(Tissue_ForDevelopementalAnalysis, sep=&quot;,&quot;)
  )</code></pre>
<p>Now let’s tidy the data by standardizing the rows, qqnorm the
columns, within each tissue.</p>
<pre class="r"><code>expression.table.tidy.qqnormed &lt;- expression.table.tidy %&gt;%
  group_by(Geneid, Tissue_ForDevelopementalAnalysis) %&gt;%
  mutate(log2expression.normalized = scale(log2expression)) %&gt;%
  mutate(var.log2expression = var(log2expression, na.rm = T)) %&gt;%
  mutate(maxExpression = max(log2expression, na.rm=T)) %&gt;%
  ungroup() %&gt;%
  group_by(ID, Tissue_ForDevelopementalAnalysis) %&gt;%
  mutate(log2expression.normalized = qnorm(dense_rank(log2expression.normalized)/(sum(!is.na(log2expression.normalized)) + 1))) %&gt;%
  ungroup()</code></pre>
<p>Let’s check PCA goes with time course, before considering linear
model (with no covariates).</p>
<pre class="r"><code>pca.results &lt;- expression.table.tidy.qqnormed %&gt;%
  filter(var.log2expression &gt; 0) %&gt;%
  filter(maxExpression &gt; 0) %&gt;%
  filter(Tissue_ForDevelopementalAnalysis == &quot;Liver&quot;) %&gt;%
  dplyr::select(ID, Geneid, log2expression.normalized) %&gt;%
  pivot_wider(names_from = &quot;ID&quot;, values_from = &quot;log2expression.normalized&quot;) %&gt;%
  drop_na() %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  t() %&gt;%
  prcomp()


pca.results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;ID&quot;) %&gt;%
  inner_join(
    CordosoSamples %&gt;%
      filter(`Used library?` %in% c(&quot;yes&quot;, &quot;Yes&quot;)) %&gt;%
      dplyr::select(ID:ID_Stage, Ordinal_stage, Tissue_ForDevelopementalAnalysis) %&gt;%
      separate_rows(Tissue_ForDevelopementalAnalysis, sep=&quot;,&quot;)) %&gt;%
  ggplot(aes(x=PC1, y=PC2, color=Ordinal_stage)) +
  geom_text(aes(label=Ordinal_stage)) +
  labs(title=&quot;PCA, liver, expression.normalized&quot;)</code></pre>
<p><img src="figure/2024-09-30_ContinousExpressionSplicingModel.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pca.results &lt;- expression.table.tidy.qqnormed %&gt;%
  filter(var.log2expression &gt; 0) %&gt;%
  filter(maxExpression &gt; 0) %&gt;%
  filter(Tissue_ForDevelopementalAnalysis == &quot;Heart&quot;) %&gt;%
  dplyr::select(ID, Geneid, log2expression.normalized) %&gt;%
  pivot_wider(names_from = &quot;ID&quot;, values_from = &quot;log2expression.normalized&quot;) %&gt;%
  drop_na() %&gt;%
  column_to_rownames(&quot;Geneid&quot;) %&gt;%
  t() %&gt;%
  prcomp()


pca.results$x %&gt;%
  as.data.frame() %&gt;%
  rownames_to_column(&quot;ID&quot;) %&gt;%
  inner_join(
    CordosoSamples %&gt;%
      filter(`Used library?` %in% c(&quot;yes&quot;, &quot;Yes&quot;)) %&gt;%
      dplyr::select(ID:ID_Stage, Ordinal_stage, Tissue_ForDevelopementalAnalysis) %&gt;%
      separate_rows(Tissue_ForDevelopementalAnalysis, sep=&quot;,&quot;)) %&gt;%
  ggplot(aes(x=PC1, y=PC2, color=Ordinal_stage)) +
  geom_text(aes(label=Ordinal_stage)) +
  labs(title=&quot;PCA, heart, expression.normalized&quot;)</code></pre>
<p><img src="figure/2024-09-30_ContinousExpressionSplicingModel.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Wow, these PCA’s actually make sense. Unlike the splicing PCAs. Maybe
there is some read depth effect going on in the splicing PCA’s. In any
case, let’s try recalculating expression spearmans in a way that might
not create all these extreme negatives which I think might be
problematic…</p>
<pre class="r"><code>spearman.coefs &lt;- expression.table.tidy.qqnormed %&gt;%
  group_by(Geneid, Tissue_ForDevelopementalAnalysis) %&gt;%
  summarise(cor.normalized = cor(log2expression.normalized, Ordinal_stage, method=&#39;s&#39;),
            cor.raw = cor(log2expression, Ordinal_stage, method=&#39;s&#39;)) %&gt;%
  ungroup()

spearman.coefs %&gt;%
  inner_join(
    expression.table.tidy.qqnormed %&gt;%
      distinct(Geneid, Tissue_ForDevelopementalAnalysis, .keep_all=T) %&gt;%
      dplyr::select(Geneid, Tissue_ForDevelopementalAnalysis, maxExpression, var.log2expression)
  ) %&gt;%
  mutate(MaxExpressionOver1 = maxExpression &gt; 1) %&gt;%
  ggplot(aes(x=cor.normalized, y=cor.raw)) +
  geom_point(alpha=0.01) +
  facet_grid(Tissue_ForDevelopementalAnalysis~MaxExpressionOver1) +
  labs(x=&quot;cor(expression_qqnorm_by_tissue, stage)&quot;, y=&quot;cor(expression, stage)&quot;)</code></pre>
<p><img src="figure/2024-09-30_ContinousExpressionSplicingModel.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>spearman.coefs %&gt;%
  inner_join(
    expression.table.tidy.qqnormed %&gt;%
      distinct(Geneid, Tissue_ForDevelopementalAnalysis, .keep_all=T) %&gt;%
      dplyr::select(Geneid, Tissue_ForDevelopementalAnalysis, maxExpression, var.log2expression)
  ) %&gt;%
  # mutate(MaxExpressionOver1 = maxExpression &gt; 1) %&gt;%
  ggplot(aes(x=cor.normalized, y=cor.raw, color=maxExpression)) +
  geom_point(alpha=0.01) +
  scale_color_viridis_c() +
  facet_wrap(~Tissue_ForDevelopementalAnalysis) +
  labs(x=&quot;cor(expression_qqnorm_by_tissue, stage)&quot;, y=&quot;cor(expression, stage)&quot;)</code></pre>
<p><img src="figure/2024-09-30_ContinousExpressionSplicingModel.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, so there is clearly a problem for the lowly expressed genes
whether you correlate by raw expression, or after qqnormed by
tissue.</p>
<p>Really, filtering by max CPM makes slightly more sense than RPKM/TPM,
since gene length isn’t so relevant in adjusting for this technical
issue of lowly <em>counted</em> genes. Nonetheless, just using a filter
of max TPM&gt;1 is probably sufficicent.</p>
<p>Let’s see if this is related to the difference in spearman with Maxin
RPKM quantifications…</p>
<pre class="r"><code>MazinStages &lt;- read_tsv(&quot;../output/CordosoMoreira_CalibratedStageTable.tsv&quot;)


RPKM.Mazin.Rabbit &lt;- Sys.glob(&quot;../code/kaessman_AS_dat/FromWebApp/rabbit/RPKM.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/kaessman_AS_dat/FromWebApp/(.+?)/RPKM.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(fread, sep=&#39; &#39;) %&gt;%
  lapply(function(x) filter(x, Names %in% spearman.coefs$Geneid)) %&gt;%
  lapply(function(x) pivot_longer(x, names_to = &quot;sample&quot;,values_to = &quot;RPKM&quot;,-Names)) %&gt;%
  bind_rows(.id=&quot;Species&quot;) %&gt;%
  mutate(sample = paste(str_to_title(Species), sample, sep=&quot;.&quot;)) %&gt;%
  separate(sample, into=c(&quot;Species&quot;, &quot;Tissue&quot;, &quot;StageName&quot;, &quot;Dummy&quot;), convert=T, remove=F) %&gt;%
  inner_join(MazinStages)


CordosoSamples %&gt;%
  dplyr::select(Species = ID_Species, StageName=Label_As_in_PreviousAnalysis, Ordinal_stage_new=Ordinal_stage) %&gt;%
  distinct() %&gt;%
  inner_join(MazinStages) %&gt;%
  ggplot(aes(x=Ordinal_stage_new, y=OrdinalStage.Manual
)) +
  geom_point() +
  facet_wrap(~Species) +
  labs(title=&quot;Check that old ordinal stage is same as new&quot;)</code></pre>
<p><img src="figure/2024-09-30_ContinousExpressionSplicingModel.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>RPKM.Mazin.Rabbit.cor &lt;- RPKM.Mazin.Rabbit %&gt;%
  group_by(Tissue, Names) %&gt;%
  summarise(cor.RPKM = cor(RPKM, OrdinalStage.Manual, method = &quot;sp&quot;)) %&gt;%
  ungroup()


# my_scatter &lt;- function(data, mapping, ...) {
#   ggplot(data = data, mapping=mapping) +
#     geom_point(..., aes(color=maxExpression), alpha=0.01) +
#     scale_color_viridis_c() +
# }

spearman.coefs %&gt;%
  inner_join(
    expression.table.tidy.qqnormed %&gt;%
      distinct(Geneid, Tissue_ForDevelopementalAnalysis, .keep_all=T) %&gt;%
      dplyr::select(Geneid, Tissue_ForDevelopementalAnalysis, maxExpression, var.log2expression)
  ) %&gt;%
  inner_join(
    RPKM.Mazin.Rabbit.cor %&gt;%
      dplyr::select(Tissue_ForDevelopementalAnalysis=Tissue, Geneid=Names, cor.MazinRPKM=cor.RPKM)
  ) %&gt;%
  ggplot(aes(x=cor.MazinRPKM, y=cor.raw, color=maxExpression)) +
  geom_point(alpha=0.01) +
  scale_color_viridis_c() +
  facet_wrap(~Tissue_ForDevelopementalAnalysis) +
  labs(x=&quot;cor(expression_Mazin, stage)&quot;, y=&quot;cor(expression, stage)&quot;)</code></pre>
<p><img src="figure/2024-09-30_ContinousExpressionSplicingModel.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>spearman.coefs %&gt;%
  inner_join(
    expression.table.tidy.qqnormed %&gt;%
      distinct(Geneid, Tissue_ForDevelopementalAnalysis, .keep_all=T) %&gt;%
      dplyr::select(Geneid, Tissue_ForDevelopementalAnalysis, maxExpression, var.log2expression)
  ) %&gt;%
  inner_join(
    RPKM.Mazin.Rabbit.cor %&gt;%
      dplyr::select(Tissue_ForDevelopementalAnalysis=Tissue, Geneid=Names, cor.MazinRPKM=cor.RPKM)
  ) %&gt;%
  ggplot(aes(x=cor.MazinRPKM, y=cor.normalized, color=maxExpression)) +
  geom_point(alpha=0.01) +
  scale_color_viridis_c() +
  facet_wrap(~Tissue_ForDevelopementalAnalysis) +
  labs(x=&quot;cor(expression_Mazing, stage)&quot;, y=&quot;cor(expression_qqnorm_by_tissue, stage)&quot;)</code></pre>
<p><img src="figure/2024-09-30_ContinousExpressionSplicingModel.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>spearman.coefs %&gt;%
  inner_join(
    expression.table.tidy.qqnormed %&gt;%
      distinct(Geneid, Tissue_ForDevelopementalAnalysis, .keep_all=T) %&gt;%
      dplyr::select(Geneid, Tissue_ForDevelopementalAnalysis, maxExpression, var.log2expression)
  ) %&gt;%
  inner_join(
    RPKM.Mazin.Rabbit.cor %&gt;%
      dplyr::select(Tissue_ForDevelopementalAnalysis=Tissue, Geneid=Names, cor.MazinRPKM=cor.RPKM)
  ) %&gt;%
  pivot_longer(names_to = &quot;stat&quot;, values_to = &quot;rho&quot;, c(&quot;cor.normalized&quot;, &quot;cor.raw&quot;, &quot;cor.MazinRPKM&quot;)) %&gt;%
  ggplot(aes(x=rho, fill=maxExpression&gt;1)) +
  geom_histogram(position = &quot;stack&quot;) +
  facet_grid(Tissue_ForDevelopementalAnalysis~stat)</code></pre>
<p><img src="figure/2024-09-30_ContinousExpressionSplicingModel.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>spearman.coefs %&gt;%
  inner_join(
    expression.table.tidy.qqnormed %&gt;%
      distinct(Geneid, Tissue_ForDevelopementalAnalysis, .keep_all=T) %&gt;%
      dplyr::select(Geneid, Tissue_ForDevelopementalAnalysis, maxExpression, var.log2expression)
  ) %&gt;%
  inner_join(
    RPKM.Mazin.Rabbit.cor %&gt;%
      dplyr::select(Tissue_ForDevelopementalAnalysis=Tissue, Geneid=Names, cor.MazinRPKM=cor.RPKM)
  ) %&gt;%
  pivot_longer(names_to = &quot;stat&quot;, values_to = &quot;rho&quot;, c(&quot;cor.normalized&quot;, &quot;cor.raw&quot;, &quot;cor.MazinRPKM&quot;)) %&gt;%
  ggplot(aes(x=rho)) +
  geom_histogram() +
  facet_grid(Tissue_ForDevelopementalAnalysis~stat)</code></pre>
<p><img src="figure/2024-09-30_ContinousExpressionSplicingModel.Rmd/unnamed-chunk-5-5.png" width="672" style="display: block; margin: auto;" />
I’m confused now why there aren’t all those extreme negatives I
previously observed in Liver, Heart, Kidney. Let’s read in the previous
expression rhos I calculated.</p>
<pre class="r"><code>Old.spearman.coefs &lt;- read_tsv(&quot;../code/MazinLeafcutterAnalysis/ExpressionSpearmanCoefs/Rabbit_UCSC.oryCun2_ensv101.log2rpkm.tsv.gz&quot;) %&gt;%
  mutate(Geneid = str_replace(Geneid, &quot;^(.+?)\\..+$&quot;, &quot;\\1&quot;))

Old.spearman.coefs %&gt;%
  ggplot(aes(x=corr)) +
  geom_histogram() +
  facet_wrap(~Tissue)</code></pre>
<p><img src="figure/2024-09-30_ContinousExpressionSplicingModel.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok from looking at my other notebook, those negative correlations are
from the subset of things that had devAS in Mazin, and also from TPM,
not RPKM.</p>
<pre class="r"><code>Old.spearman.coefs &lt;- read_tsv(&quot;../code/MazinLeafcutterAnalysis/ExpressionSpearmanCoefs/Rabbit_UCSC.oryCun2_ensv101.log2tpm.tsv.gz&quot;) %&gt;%
  mutate(Geneid = str_replace(Geneid, &quot;^(.+?)\\..+$&quot;, &quot;\\1&quot;))

Old.spearman.coefs %&gt;%
  ggplot(aes(x=corr)) +
  geom_histogram() +
  facet_wrap(~Tissue)</code></pre>
<p><img src="figure/2024-09-30_ContinousExpressionSplicingModel.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok, wow I think the TPM is the issue.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] magrittr_2.0.3    data.table_1.14.2 forcats_0.5.1     stringr_1.4.0    
 [5] dplyr_1.0.9       purrr_0.3.4       readr_2.1.2       tidyr_1.2.0      
 [9] tibble_3.1.7      ggplot2_3.3.6     tidyverse_1.3.1  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.12       lubridate_1.8.0   assertthat_0.2.1  rprojroot_2.0.3  
 [5] digest_0.6.29     utf8_1.2.2        R6_2.5.1          cellranger_1.1.0 
 [9] backports_1.4.1   reprex_2.0.1      evaluate_0.15     highr_0.9        
[13] httr_1.4.3        pillar_1.7.0      rlang_1.0.2       readxl_1.4.0     
[17] rstudioapi_0.13   jquerylib_0.1.4   R.oo_1.24.0       R.utils_2.11.0   
[21] rmarkdown_2.14    labeling_0.4.2    bit_4.0.4         munsell_0.5.0    
[25] broom_0.8.0       compiler_4.2.0    httpuv_1.6.5      modelr_0.1.8     
[29] xfun_0.30         pkgconfig_2.0.3   htmltools_0.5.2   tidyselect_1.1.2 
[33] workflowr_1.7.0   viridisLite_0.4.0 fansi_1.0.3       crayon_1.5.1     
[37] tzdb_0.3.0        dbplyr_2.1.1      withr_2.5.0       later_1.3.0      
[41] R.methodsS3_1.8.1 grid_4.2.0        jsonlite_1.8.0    gtable_0.3.0     
[45] lifecycle_1.0.1   DBI_1.1.2         git2r_0.30.1      scales_1.3.0     
[49] vroom_1.5.7       cli_3.6.2         stringi_1.7.6     farver_2.1.0     
[53] fs_1.5.2          promises_1.2.0.1  xml2_1.3.3        bslib_0.3.1      
[57] ellipsis_0.3.2    generics_0.1.2    vctrs_0.4.1       tools_4.2.0      
[61] bit64_4.0.5       glue_1.6.2        hms_1.1.1         parallel_4.2.0   
[65] fastmap_1.1.0     yaml_2.3.5        colorspace_2.0-3  rvest_1.0.2      
[69] knitr_1.39        haven_2.5.0       sass_0.4.1       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
