<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-10-15" />

<title>2024-10-15_HeatmapsWithLeafcutter_dPSI</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2024_ComparativeSplicing</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">2024-10-15_HeatmapsWithLeafcutter_dPSI</h1>
<h4 class="date">2024-10-15</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#intro" id="toc-intro">Intro</a></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-16
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>2024_comparativesplicing/analysis/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed19900924code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(19900924)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed19900924code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(19900924)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcombfairkun2024comparativesplicingtree172f74b99c3d796bd47fa2cac0f2aa13fcd77647targetblank172f74ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/bfairkun/2024_comparativesplicing/tree/172f74b99c3d796bd47fa2cac0f2aa13fcd77647" target="_blank">172f74b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcombfairkun2024comparativesplicingtree172f74b99c3d796bd47fa2cac0f2aa13fcd77647targetblank172f74ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/bfairkun/2024_comparativesplicing/tree/172f74b99c3d796bd47fa2cac0f2aa13fcd77647" target="_blank">172f74b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.DS_Store
    Ignored:    code/.RData
    Ignored:    code/.Rhistory
    Ignored:    code/.ipynb_checkpoints/
    Ignored:    code/.snakemake/
    Ignored:    code/ChainFiles/
    Ignored:    code/CordosoMoreira_Fastq/
    Ignored:    code/Downloads/
    Ignored:    code/GenomeFiles/
    Ignored:    code/LiftoverJuncs/
    Ignored:    code/Log.out
    Ignored:    code/MazinLeafcutterAnalysis/
    Ignored:    code/Rplots.pdf
    Ignored:    code/Session.vim
    Ignored:    code/config/OldConfigs/2040822_Cordoso_Moreira_SampleList.tsv
    Ignored:    code/conservation/
    Ignored:    code/featureCounts/
    Ignored:    code/kaessmanAnalysis/
    Ignored:    code/kaessman_AS_dat/
    Ignored:    code/logs/
    Ignored:    code/rna-seq/
    Ignored:    code/rules/.MazinLeafcutterAnalysis.smk.swp
    Ignored:    code/scratch/
    Ignored:    code/scripts/.ipynb_checkpoints/
    Ignored:    code/scripts/.vscode/
    Ignored:    code/snakemake.log
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/2024-08-21_SpearmanFromPSI_WithinSpecies.Rmd
    Untracked:  analysis/2024-08-21_SpearmanFromPSI_WithinSpecies_AllSpecies.Rmd
    Untracked:  analysis/2024-08-21_SpearmanFromPSI_WithinSpecies_SampleStagesFixed.Rmd
    Untracked:  analysis/2024-08-24_BioMartLookupGenes.Rmd
    Untracked:  analysis/2024-08-29_ExploreJuncLiftovers.Rmd
    Untracked:  analysis/2024-08-29_OrganizeCrossSpeciesTables.Rmd
    Untracked:  analysis/2024-09-06_UnrpdoctuviveSplicingAndExpression.Rmd
    Untracked:  analysis/2024-09-09_Organize_ConserveddevASJuncs.Rmd
    Untracked:  analysis/2024-09-10_RedoMazinJuncAnalysisFixedClassifications.Rmd
    Untracked:  analysis/2024-09-23_FixMouseEnsemblBed12ToGtf.Rmd
    Untracked:  analysis/2024-09-30_ContinousExpressionSplicingModel.Rmd
    Untracked:  analysis/2024-09-30_ContinousTimeSplicingModel.Rmd
    Untracked:  analysis/2024-10-03_ExplorePuzzling_UP_vs_Expression_Effects.Rmd
    Untracked:  analysis/2024-10-14_MakeLeafcutterContrasts.Rmd
    Untracked:  analysis/2024-10-15_HeatmapsWithLeafcutter_dPSI.Rmd
    Untracked:  analysis/20240815_LiftoverJuncsTest.ipynb
    Untracked:  analysis/Untitled.ipynb
    Untracked:  code/config/CordosoGenomes_Extra_Gtfs.tsv
    Untracked:  code/config/CordosoTimeSeriesContrasts.tsv
    Untracked:  code/config/GTEx_juncFileList.tsv
    Untracked:  code/envs/crossmap.yml
    Untracked:  code/envs/py27.yml
    Untracked:  code/rules/MazinLeafcutterAnalysis.smk
    Untracked:  code/scripts/FeatureCounts_to_Mat.R
    Untracked:  code/scripts/PrepAllJuncsFor_JunctionClassifier.R
    Untracked:  code/scripts/QQNorm_PSITable_ByTissue.R
    Untracked:  code/scripts/SpearmanCor_Mazin_LeafcutterPSI.R
    Untracked:  code/scripts/SpearmanCor_Mazin_log2RPKM.R
    Untracked:  code/scripts/Untitled.ipynb
    Untracked:  code/scripts/leafcutter_to_PSI_GTEX.R
    Untracked:  code/scripts/tidy_leafcutter_ds_results.R
    Untracked:  data/Stages_AsIn_CordosoMoreira.tsv
    Untracked:  data/Stages_AsIn_CordosoMoreira_Recoded.txt
    Untracked:  output/Conserved.devAS.leafcutter.tsv.gz
    Untracked:  output/Ensembl.GeneHumanHomologs.tsv.gz
    Untracked:  output/Ensembl.TranscriptInfo.tsv.gz
    Untracked:  output/GTEx_DS_DE_FromChao_CordosoTissuePairs.tsv.gz

Unstaged changes:
    Modified:   analysis/2024-07-16_Download_CordosoMoreira_Fastq.Rmd
    Modified:   analysis/index.Rmd
    Modified:   code/Snakefile
    Modified:   code/config/ChainFiles.tsv
    Modified:   code/config/Cordoso_Moreira_SampleList.tsv
    Modified:   code/config/STAR_Genome_List.tsv
    Modified:   code/config/samples.tsv
    Modified:   code/envs/bedparse.yml
    Modified:   code/module_workflows/snakemake-workflow_rna-seq
    Modified:   code/rules/LiftoverJuncs.smk
    Modified:   code/rules/common.smk
    Modified:   code/scripts/daiuc_leafcutter2
    Modified:   code/scripts/leafcutter2
    Modified:   output/QC/ReadCountsPerSamples.tsv

Staged changes:
    Modified:   .gitmodules
    New:        code/scripts/daiuc_leafcutter2

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="intro" class="section level2">
<h2>Intro</h2>
<p>Haven’t been able to figure out why unproductive splicing as measured
by spearman of leafcutter PSI doesn’t correlate negatively with spearman
of host gene expression. Now, rather than using spearman correlation
coefficients, I am measuring logFC and dPSI (from leafcutter) using
contrasts from groups of samples at the beginning or end of the the
species:tissue time course.</p>
<pre class="r"><code>library(tidyverse)
library(data.table)

# Set theme
theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family=&quot;Helvetica&quot;)))

# I use layer a lot, to rotate long x-axis labels
Rotate_x_labels &lt;- theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))</code></pre>
<p>Now read in the data… I defined the groups for leafcutter and edgeR
contrasts a couple different ways, as described in previous
notebook.</p>
<pre class="r"><code>CordosoSamples &lt;- read_tsv(&quot;../code/config/Cordoso_Moreira_SampleList.tsv&quot;)

dat &lt;- Sys.glob(&quot;../code/MazinLeafcutterAnalysis/Contrasts_ds_tidy/*joined.tsv.gz&quot;) %&gt;%
  setNames(str_replace(., &quot;../code/MazinLeafcutterAnalysis/Contrasts_ds_tidy/(.+?).joined.tsv.gz&quot;, &quot;\\1&quot;)) %&gt;%
  lapply(fread) %&gt;%
  bind_rows(.id=&quot;Contrast&quot;) %&gt;%
  separate(Contrast, into=c(&quot;Contrast&quot;, &quot;Tissue&quot;, &quot;Species&quot;), sep=&quot;\\.&quot;) %&gt;%
  mutate(IntFlag = 1*UTR + 2*Coding + 4*Annot + 8*GencodePC) %&gt;%
  mutate(ProductivityLabel = case_when(
    IntFlag %in% c(1,5) ~ &quot;NE&quot;,
    IntFlag %in% c(0,4) ~ &quot;UP&quot;,
    TRUE ~ &quot;PR&quot;
  ))


dat %&gt;%
  count(Contrast, Tissue, Species)</code></pre>
<pre><code>                    Contrast Tissue Species      n
  1:        BeforeB_vs_AferS  Brain Chicken  84424
  2:        BeforeB_vs_AferS  Brain   Human 173105
  3:        BeforeB_vs_AferS  Brain Macaque 112016
  4:        BeforeB_vs_AferS  Brain   Mouse 117412
  5:        BeforeB_vs_AferS  Brain Opossum  73956
 ---                                              
218: FirstHalf_vs_SecondHalf Testis Macaque 142102
219: FirstHalf_vs_SecondHalf Testis   Mouse 129304
220: FirstHalf_vs_SecondHalf Testis Opossum 110314
221: FirstHalf_vs_SecondHalf Testis  Rabbit 141994
222: FirstHalf_vs_SecondHalf Testis     Rat 147598</code></pre>
<pre class="r"><code>dat %&gt;%
  filter(p.adjust &lt; 0.1) %&gt;%
  filter(abs(logef)&gt;1) %&gt;%
  filter(abs(deltapsi) &gt; 0.1) %&gt;%
    # count(Contrast, Tissue, Species)
  group_by(Contrast, Tissue, Species, ProductivityLabel) %&gt;%
  summarise(spearman.of_CorPSI_vs_CorRPKM = cor(deltapsi, logFC, method=&#39;s&#39;, use=&quot;pairwise.complete.obs&quot;), n=n()) %&gt;%
  ungroup() %&gt;%
  ggplot(aes(x=Species, y=Tissue, fill=spearman.of_CorPSI_vs_CorRPKM)) +
  geom_tile() +
  scale_fill_gradient2(midpoint=0) +
  geom_text(aes(label=n), size=2) +
  facet_grid(Contrast~ProductivityLabel) +
  Rotate_x_labels</code></pre>
<p><img src="figure/2024-10-15_HeatmapsWithLeafcutter_dPSI.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&quot;../code/scratch/HeatmapFromLeafcutterDpsi_and_log2FC.pdf&quot;, height=12, width=16)

dat %&gt;%
  count(Species, Contrast)</code></pre>
<pre><code>    Species                Contrast       n
 1: Chicken        BeforeB_vs_AferS  561476
 2: Chicken       BeforeB_vs_AfterB  625347
 3: Chicken       BeforeM_vs_AfterS  533933
 4: Chicken     First10_vs_Second10  571129
 5: Chicken FirstHalf_vs_SecondHalf  653636
 6:   Human        BeforeB_vs_AferS  719684
 7:   Human       BeforeB_vs_AfterB  918954
 8:   Human       BeforeM_vs_AfterS  671503
 9:   Human     First10_vs_Second10  958734
10:   Human FirstHalf_vs_SecondHalf 1252650
11: Macaque        BeforeB_vs_AferS  620277
12: Macaque       BeforeB_vs_AfterB  705032
13: Macaque     First10_vs_Second10  636292
14: Macaque FirstHalf_vs_SecondHalf  724478
15:   Mouse        BeforeB_vs_AferS  753926
16:   Mouse       BeforeB_vs_AfterB  829980
17:   Mouse       BeforeM_vs_AfterS  471246
18:   Mouse     First10_vs_Second10  672340
19:   Mouse FirstHalf_vs_SecondHalf  892190
20: Opossum        BeforeB_vs_AferS  298451
21: Opossum       BeforeB_vs_AfterB  384640
22: Opossum       BeforeM_vs_AfterS  681044
23: Opossum     First10_vs_Second10  652553
24: Opossum FirstHalf_vs_SecondHalf  744004
25:  Rabbit        BeforeB_vs_AferS  849876
26:  Rabbit       BeforeB_vs_AfterB  926683
27:  Rabbit       BeforeM_vs_AfterS  849876
28:  Rabbit     First10_vs_Second10  779293
29:  Rabbit FirstHalf_vs_SecondHalf 1033056
30:     Rat        BeforeB_vs_AferS  884427
31:     Rat       BeforeB_vs_AfterB  981411
32:     Rat       BeforeM_vs_AfterS  782111
33:     Rat     First10_vs_Second10  738769
34:     Rat FirstHalf_vs_SecondHalf 1020611
    Species                Contrast       n</code></pre>
<p>Let’s look at specific cell at beta beta scatter.</p>
<p>First let’s verify I understand the signs of the logFC and deltapsi
correctly… I believe it should be Late/Early (so that positive deltapsi
means increase in PSI over time, and positive logFC means increase in
expression over time)</p>
<pre class="r"><code>dat %&gt;%
  filter(p.adjust &lt; 0.1) %&gt;%
  filter(abs(logef)&gt;1) %&gt;%
  filter(abs(deltapsi) &gt; 0.1) %&gt;%
  filter(Contrast == &quot;BeforeM_vs_AfterS&quot; &amp; Species == &quot;Rat&quot; &amp; Tissue == &quot;Brain&quot;) %&gt;%
  filter(FDR&lt;0.001) %&gt;%
  arrange(desc(abs(logFC))) %&gt;%
  head()</code></pre>
<pre><code>            Contrast Tissue Species           cluster     start      stop
1: BeforeM_vs_AfterS  Brain     Rat  chrX:clu_65817_+ 100195281 100196341
2: BeforeM_vs_AfterS  Brain     Rat  chrX:clu_65817_+ 100196509 100197119
3: BeforeM_vs_AfterS  Brain     Rat chr18:clu_29151_+  54417539  54423551
4: BeforeM_vs_AfterS  Brain     Rat  chr8:clu_59492_+  90502371  90502804
5: BeforeM_vs_AfterS  Brain     Rat  chr7:clu_54409_-  12012292  12013384
6: BeforeM_vs_AfterS  Brain     Rat chr15:clu_22718_+  45151511  45152330
       logef        Early      Late   deltapsi             Intron_coord  status
1:  3.098595 7.825390e-03 0.3332104  0.3253850 chrX:100195281-100196340 Success
2: -1.050253 9.007364e-01 0.6053251 -0.2954114 chrX:100196509-100197118 Success
3:  1.387213 5.690950e-01 0.9136454  0.3445503  chr18:54417539-54423550 Success
4:  3.647285 3.841176e-01 0.4848078  0.1006902   chr8:90502371-90502803 Success
5:  9.131374 2.891287e-07 0.2043488  0.2043485   chr7:12012292-12013383 Success
6:  1.443385 4.314378e-03 0.1301252  0.1258108  chr15:45151511-45152329 Success
       loglr df            p     p.adjust Gene_name Strand Annot Coding   UTR
1:  53.67680  3 4.071925e-23 2.007720e-21      Plp1      +  TRUE   TRUE FALSE
2:  53.67680  3 4.071925e-23 2.007720e-21      Plp1      +  TRUE   TRUE FALSE
3:  51.98334  2 2.654144e-23 1.326867e-21    Camk2a      +  TRUE   TRUE FALSE
4:  12.11374  2 5.483662e-06 4.556102e-05   Rasgrf1      +  TRUE   TRUE  TRUE
5:  19.73551  2 2.685207e-09 3.619974e-08    Cyp4f1      -  TRUE   TRUE FALSE
6: 100.61250  4 2.048774e-42 3.968119e-40      Egr3      +  TRUE   TRUE FALSE
   GencodePC    logFC     logCPM       PValue          FDR IntFlag
1:      TRUE 7.942555 10.0330790 0.000000e+00 0.000000e+00      14
2:      TRUE 7.942555 10.0330790 0.000000e+00 0.000000e+00      14
3:      TRUE 6.825270  9.9117867 4.701417e-68 7.965701e-67      14
4:      TRUE 6.452353  7.2140849 1.650059e-17 4.693253e-17      15
5:      TRUE 6.211570  0.5229616 1.131230e-76 2.461529e-75      14
6:      TRUE 5.832464  5.0623401 2.315607e-73 4.636306e-72      14
   ProductivityLabel
1:                PR
2:                PR
3:                PR
4:                PR
5:                PR
6:                PR</code></pre>
<pre class="r"><code>RPKM.rat &lt;- fread(&quot;../code/MazinLeafcutterAnalysis/Expression/Rat_UCSC.rn7_RefSeqv108.log2rpkm.tsv.gz&quot;)

RPKM.rat %&gt;%
  filter(Geneid == &quot;Plp1&quot;) %&gt;%
  pivot_longer(names_to = &quot;ID&quot;, values_to = &quot;log2RPKM&quot;, -Geneid) %&gt;%
  inner_join(
    CordosoSamples %&gt;%
      filter(ID_Species == &quot;Rat&quot;) %&gt;%
      filter(`Used library?` %in% c(&quot;Yes&quot;, &quot;yes&quot;)) %&gt;%
      separate_rows(Tissue_ForDevelopementalAnalysis) %&gt;%
      filter(Tissue_ForDevelopementalAnalysis == &quot;Brain&quot;)
  ) %&gt;%
  ggplot(aes(x=Ordinal_stage, y=log2RPKM)) +
  geom_point(alpha=0.3) +
  geom_smooth(method=&#39;loess&#39;, se=F)</code></pre>
<p><img src="figure/2024-10-15_HeatmapsWithLeafcutter_dPSI.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Ok I understand the effect directions properly. Now let’s look at one
of the scatters…</p>
<pre class="r"><code>dat %&gt;%
  filter(p.adjust &lt; 0.1) %&gt;%
  filter(abs(logef)&gt;1) %&gt;%
  filter(abs(deltapsi) &gt; 0.1) %&gt;%
  filter(Contrast == &quot;BeforeM_vs_AfterS&quot; &amp; Species == &quot;Rat&quot; &amp; Tissue == &quot;Brain&quot;) %&gt;%
  ggplot(aes(x=deltapsi, y=logFC)) +
  geom_point(alpha=0.2) +
  geom_smooth(method=&#39;lm&#39;) +
  facet_wrap(~ProductivityLabel)</code></pre>
<p><img src="figure/2024-10-15_HeatmapsWithLeafcutter_dPSI.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now let’s play with different thresholds for deltapsi and logef</p>
<pre class="r"><code>dat$Contrast %&gt;% unique</code></pre>
<pre><code>[1] &quot;BeforeB_vs_AferS&quot;        &quot;BeforeB_vs_AfterB&quot;      
[3] &quot;BeforeM_vs_AfterS&quot;       &quot;First10_vs_Second10&quot;    
[5] &quot;FirstHalf_vs_SecondHalf&quot;</code></pre>
<pre class="r"><code>dat %&gt;%
  filter(p.adjust &lt; 0.05) %&gt;%
  # filter(abs(logef)&gt;0.5) %&gt;%
  filter(abs(deltapsi) &gt; 0.01) %&gt;%
  filter(!ProductivityLabel==&quot;NE&quot;) %&gt;%
  group_by(Contrast, Species, Tissue, cluster) %&gt;%
  mutate(Contains_Noncoding = any(ProductivityLabel == &quot;UP&quot;)) %&gt;%
  ungroup() %&gt;%
  mutate(ProductivityLabel = factor(ProductivityLabel, levels=c(&quot;UP&quot;, &quot;PR&quot;))) %&gt;%
  arrange(Contrast, Species, Tissue, cluster, Contains_Noncoding, ProductivityLabel, desc(abs(deltapsi))) %&gt;%
  distinct(Contrast, Species, Tissue, cluster, .keep_all=T) %&gt;%
    # count(Contrast, Tissue, Species)
  group_by(Contrast, Tissue, Species, ProductivityLabel) %&gt;%
  summarise(spearman.of_dPSI_vs_logFC = cor(deltapsi, logFC, method=&#39;s&#39;, use=&quot;pairwise.complete.obs&quot;), n=n()) %&gt;%
  ungroup() %&gt;%
  mutate(Contrast = recode(Contrast, &quot;BeforeB_vs_AferS&quot;=&quot;Before birth vs\nafter spermatogenesis&quot;, &quot;BeforeB_vs_AfterB&quot;=&quot;Before birth vs\nafter birth&quot;, &quot;BeforeM_vs_AfterS&quot;=&quot;Before oogenesis vs\nafter birth&quot;, &quot;First10_vs_Second10&quot;=&quot;First 10 samples vs\n last 10 samples&quot;, &quot;FirstHalf_vs_SecondHalf&quot;=&quot;First half vs\nlatter half&quot;)) %&gt;%
  ggplot(aes(x=Species, y=Tissue, fill=spearman.of_dPSI_vs_logFC)) +
  geom_tile() +
  scale_fill_gradient2(midpoint=0) +
  geom_text(aes(label=n), size=2) +
  facet_grid(ProductivityLabel~Contrast) +
  Rotate_x_labels +
  theme(legend.position=&#39;bottom&#39;) +
  labs(caption = &quot;One junc per cluster in each tissue:species pair included.\nleafcutter_ds p.adjust &lt; 0.05 and abs(deltapsi) &gt; 1%&quot;)</code></pre>
<p><img src="figure/2024-10-15_HeatmapsWithLeafcutter_dPSI.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;../code/scratch/HeatmapFromLeafcutterDpsi_and_log2FC.pdf&quot;, height=8, width=14)


dat %&gt;%
  filter(p.adjust &lt; 0.05) %&gt;%
  # filter(abs(logef)&gt;0.5) %&gt;%
  filter(abs(deltapsi) &gt; 0.05) %&gt;%
  filter(!ProductivityLabel==&quot;NE&quot;) %&gt;%
  group_by(Contrast, Species, Tissue, cluster) %&gt;%
  mutate(Contains_Noncoding = any(ProductivityLabel == &quot;UP&quot;)) %&gt;%
  ungroup() %&gt;%
  mutate(ProductivityLabel = factor(ProductivityLabel, levels=c(&quot;UP&quot;, &quot;PR&quot;))) %&gt;%
  arrange(Contrast, Species, Tissue, cluster, Contains_Noncoding, ProductivityLabel, desc(abs(deltapsi))) %&gt;%
  distinct(Contrast, Species, Tissue, cluster, .keep_all=T) %&gt;%
    # count(Contrast, Tissue, Species)
  group_by(Contrast, Tissue, Species, ProductivityLabel) %&gt;%
  summarise(spearman.of_dPSI_vs_logFC = cor(deltapsi, logFC, method=&#39;s&#39;, use=&quot;pairwise.complete.obs&quot;), n=n()) %&gt;%
  ungroup() %&gt;%
  mutate(Contrast = recode(Contrast, &quot;BeforeB_vs_AferS&quot;=&quot;Before birth vs\nafter spermatogenesis&quot;, &quot;BeforeB_vs_AfterB&quot;=&quot;Before birth vs\nafter birth&quot;, &quot;BeforeM_vs_AfterS&quot;=&quot;Before oogenesis vs\nafter birth&quot;, &quot;First10_vs_Second10&quot;=&quot;First 10 samples vs\n last 10 samples&quot;, &quot;FirstHalf_vs_SecondHalf&quot;=&quot;First half vs\nlatter half&quot;)) %&gt;%
  ggplot(aes(x=Species, y=Tissue, fill=spearman.of_dPSI_vs_logFC)) +
  geom_tile() +
  scale_fill_gradient2(midpoint=0) +
  geom_text(aes(label=n), size=2) +
  facet_grid(ProductivityLabel~Contrast) +
  Rotate_x_labels +
  theme(legend.position=&#39;bottom&#39;) +
  labs(caption = &quot;One junc per cluster in each tissue:species pair included.\nleafcutter_ds p.adjust &lt; 0.05 and abs(deltapsi) &gt; 5%&quot;)</code></pre>
<p><img src="figure/2024-10-15_HeatmapsWithLeafcutter_dPSI.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;../code/scratch/HeatmapFromLeafcutterDpsi_and_log2FC.stricterfiltering.pdf&quot;, height=8, width=14)</code></pre>
<p>Also, let’s present more similar to Chao has:</p>
<pre class="r"><code>dat %&gt;%
  filter(p.adjust &lt; 0.05) %&gt;%
  # filter(abs(logef)&gt;0.5) %&gt;%
  filter(abs(deltapsi) &gt; 0.05) %&gt;%
  filter(!ProductivityLabel==&quot;NE&quot;) %&gt;%
  group_by(Contrast, Species, Tissue, cluster) %&gt;%
  mutate(Contains_Noncoding = any(ProductivityLabel == &quot;UP&quot;)) %&gt;%
  ungroup() %&gt;%
  mutate(ProductivityLabel = factor(ProductivityLabel, levels=c(&quot;UP&quot;, &quot;PR&quot;))) %&gt;%
  arrange(Contrast, Species, Tissue, cluster, Contains_Noncoding, ProductivityLabel, desc(abs(deltapsi))) %&gt;%
  distinct(Contrast, Species, Tissue, cluster, .keep_all=T) %&gt;%
  # count(Contrast, Tissue, Species)
  group_by(Contrast, Tissue, Species, ProductivityLabel) %&gt;%
  summarise(spearman.of_dPSI_vs_logFC = cor(deltapsi, logFC, method=&#39;s&#39;, use=&quot;pairwise.complete.obs&quot;), n=n(),
            p  = cor.test(deltapsi, logFC, method=&#39;s&#39;, use=&quot;pairwise.complete.obs&quot;)$p.value) %&gt;%
  ungroup() %&gt;%
  filter(Contrast == &quot;First10_vs_Second10&quot;) %&gt;%
  ggplot(aes(x=spearman.of_dPSI_vs_logFC, y=-log10(p))) +
  geom_point(aes(color=Species, shape=Tissue)) +
  geom_hline(yintercept = 3, linetype=&#39;dashed&#39;) +
  facet_wrap(~ProductivityLabel) +
  coord_cartesian(ylim=c(0,10))</code></pre>
<p><img src="figure/2024-10-15_HeatmapsWithLeafcutter_dPSI.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" />
Ok, let’s start making some final looking plots…</p>
<pre class="r"><code>Final.heatmap.dat &lt;- dat %&gt;%
  filter(p.adjust &lt; 0.05) %&gt;%
  # filter(abs(logef)&gt;0.5) %&gt;%
  filter(abs(deltapsi) &gt; 0.05) %&gt;%
  filter(!ProductivityLabel==&quot;NE&quot;) %&gt;%
  filter(Contrast == &quot;First10_vs_Second10&quot;) %&gt;%
  mutate(Contrast = recode(Contrast, &quot;BeforeB_vs_AferS&quot;=&quot;Before birth vs\nafter spermatogenesis&quot;, &quot;BeforeB_vs_AfterB&quot;=&quot;Before birth vs\nafter birth&quot;, &quot;BeforeM_vs_AfterS&quot;=&quot;Before oogenesis vs\nafter birth&quot;, &quot;First10_vs_Second10&quot;=&quot;First 10 samples vs\n last 10 samples&quot;, &quot;FirstHalf_vs_SecondHalf&quot;=&quot;First half vs\nlatter half&quot;)) %&gt;%
  group_by(Contrast, Species, Tissue, cluster) %&gt;%
  mutate(Contains_Noncoding = any(ProductivityLabel == &quot;UP&quot;)) %&gt;%
  ungroup() %&gt;%
  mutate(ProductivityLabel = factor(ProductivityLabel, levels=c(&quot;UP&quot;, &quot;PR&quot;))) %&gt;%
  arrange(Contrast, Species, Tissue, cluster, Contains_Noncoding, ProductivityLabel, desc(abs(deltapsi))) %&gt;%
  distinct(Contrast, Species, Tissue, cluster, .keep_all=T)

Final.heatmap.dat %&gt;%
  group_by(Contrast, Tissue, Species, ProductivityLabel) %&gt;%
  summarise(spearman.of_dPSI_vs_logFC = cor(deltapsi, logFC, method=&#39;s&#39;, use=&quot;pairwise.complete.obs&quot;), n=n(),
            p  = cor.test(deltapsi, logFC, method=&#39;s&#39;, use=&quot;pairwise.complete.obs&quot;)$p.value) %&gt;%
  ungroup() %&gt;%
  mutate(Plab = case_when(
    p &gt; 0.05 ~ &quot;&quot;,
    p &lt; 0.05 ~ &quot;*&quot;,
    p &lt; 0.005 ~ &quot;**&quot;,
    p &lt; 0.0005 ~ &quot;***&quot;
  )) %&gt;%
  mutate(label = paste0(n, Plab)) %&gt;%
  mutate(Species = factor(Species, levels=c(&quot;Human&quot;, &quot;Macaque&quot;, &quot;Mouse&quot;, &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Opossum&quot;, &quot;Chicken&quot;))) %&gt;%
  ggplot(aes(x=Species, y=Tissue, fill=spearman.of_dPSI_vs_logFC)) +
  geom_tile() +
  scale_fill_gradient2(midpoint=0) +
  geom_text(aes(label=label), size=2) +
  facet_wrap(~ProductivityLabel, nrow=2) +
  Rotate_x_labels +
  theme() +
  labs(caption = &quot;One junc per cluster in each tissue:species pair included.\nleafcutter_ds p.adjust &lt; 0.05 and abs(deltapsi) &gt; 5%&quot;, fill=&quot;spearman rho\ndPSI v logFC&quot;)</code></pre>
<p><img src="figure/2024-10-15_HeatmapsWithLeafcutter_dPSI.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Final.heatmap.dat %&gt;%
  group_by(Contrast, Tissue, Species, ProductivityLabel) %&gt;%
  summarise(spearman.of_dPSI_vs_logFC = cor(deltapsi, logFC, method=&#39;s&#39;, use=&quot;pairwise.complete.obs&quot;), n=n(),
            p  = cor.test(deltapsi, logFC, method=&#39;s&#39;, use=&quot;pairwise.complete.obs&quot;)$p.value) %&gt;%
  ungroup() %&gt;%
  mutate(Plab = case_when(
    p &gt; 0.05 ~ &quot;&quot;,
    p &lt; 0.05 ~ &quot;*&quot;,
    p &lt; 0.005 ~ &quot;**&quot;,
    p &lt; 0.0005 ~ &quot;***&quot;
  )) %&gt;%
  mutate(label = paste0(n, Plab)) %&gt;%
  mutate(Species = factor(Species, levels=c(&quot;Human&quot;, &quot;Macaque&quot;, &quot;Mouse&quot;, &quot;Rat&quot;, &quot;Rabbit&quot;, &quot;Opossum&quot;, &quot;Chicken&quot;))) %&gt;%
  ggplot(aes(x=Species, y=Tissue, fill=spearman.of_dPSI_vs_logFC)) +
  geom_tile() +
  scale_fill_gradient2(midpoint=0) +
  geom_text(aes(label=n), size=2) +
  facet_wrap(~ProductivityLabel, nrow=2) +
  Rotate_x_labels +
  theme() +
  labs(caption = &quot;One junc per cluster in each tissue:species pair included.\nleafcutter_ds p.adjust &lt; 0.05 and abs(deltapsi) &gt; 5%&quot;, fill=&quot;spearman rho\ndPSI v logFC&quot;)</code></pre>
<p><img src="figure/2024-10-15_HeatmapsWithLeafcutter_dPSI.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Final.heatmap.dat %&gt;%
  filter(Species == &quot;Chicken&quot; &amp; Tissue == &quot;Brain&quot;) %&gt;%
  ggplot(aes(y=deltapsi, x=logFC)) +
  geom_point(alpha=0.2) +
  facet_wrap(~ProductivityLabel) +
  labs(x=&quot;Differential gene expression (log2FC)&quot;)</code></pre>
<p><img src="figure/2024-10-15_HeatmapsWithLeafcutter_dPSI.Rmd/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Final.heatmap.dat %&gt;%
  filter(Species == &quot;Chicken&quot; &amp; Tissue == &quot;Brain&quot;) %&gt;%
  ggplot(aes(y=deltapsi, x=logFC)) +
  geom_point(alpha=0.2) +
  facet_wrap(~ProductivityLabel) +
  labs(x=&quot;Differential gene expression (log2FC)&quot;)</code></pre>
<p><img src="figure/2024-10-15_HeatmapsWithLeafcutter_dPSI.Rmd/unnamed-chunk-7-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;../code/scratch/HeatmapFromLeafcutterDpsi_and_log2FC.stricterfiltering.pdf&quot;, height=8, width=14)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] data.table_1.14.2 forcats_0.5.1     stringr_1.4.0     dplyr_1.0.9      
 [5] purrr_0.3.4       readr_2.1.2       tidyr_1.2.0       tibble_3.1.7     
 [9] ggplot2_3.3.6     tidyverse_1.3.1  

loaded via a namespace (and not attached):
 [1] httr_1.4.3        sass_0.4.1        bit64_4.0.5       vroom_1.5.7      
 [5] jsonlite_1.8.0    splines_4.2.0     R.utils_2.11.0    modelr_0.1.8     
 [9] bslib_0.3.1       assertthat_0.2.1  highr_0.9         cellranger_1.1.0 
[13] yaml_2.3.5        lattice_0.20-45   pillar_1.7.0      backports_1.4.1  
[17] glue_1.6.2        digest_0.6.29     promises_1.2.0.1  rvest_1.0.2      
[21] colorspace_2.0-3  Matrix_1.5-3      htmltools_0.5.2   httpuv_1.6.5     
[25] R.oo_1.24.0       pkgconfig_2.0.3   broom_0.8.0       haven_2.5.0      
[29] scales_1.3.0      later_1.3.0       tzdb_0.3.0        git2r_0.30.1     
[33] mgcv_1.8-40       generics_0.1.2    farver_2.1.0      ellipsis_0.3.2   
[37] withr_2.5.0       cli_3.6.2         magrittr_2.0.3    crayon_1.5.1     
[41] readxl_1.4.0      evaluate_0.15     R.methodsS3_1.8.1 fs_1.5.2         
[45] fansi_1.0.3       nlme_3.1-157      xml2_1.3.3        textshaping_0.3.6
[49] tools_4.2.0       hms_1.1.1         lifecycle_1.0.1   munsell_0.5.0    
[53] reprex_2.0.1      compiler_4.2.0    jquerylib_0.1.4   systemfonts_1.0.4
[57] rlang_1.0.2       grid_4.2.0        rstudioapi_0.13   labeling_0.4.2   
[61] rmarkdown_2.14    gtable_0.3.0      DBI_1.1.2         R6_2.5.1         
[65] lubridate_1.8.0   knitr_1.39        fastmap_1.1.0     bit_4.0.4        
[69] utf8_1.2.2        workflowr_1.7.0   rprojroot_2.0.3   ragg_1.2.5       
[73] stringi_1.7.6     parallel_4.2.0    Rcpp_1.0.12       vctrs_0.4.1      
[77] dbplyr_2.1.1      tidyselect_1.1.2  xfun_0.30        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
