---
title: "2024-08-01_PlotsForLabMeeting"
output: html_document
date: '2024-08-01'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = F, message = F)
```

## Intro

a few plots for lab meeting:

- example conserved devAS event from Mazin et al
- Number devAS events by species, with symmetry
- beta beta scatter example, with NMDFInderB approach

### read data

```{r}
library(tidyverse)
library(data.table)
library(biomaRt)


sample_n_of <- function(data, size, ...) {
  dots <- quos(...)
  
  group_ids <- data %>% 
    group_by(!!! dots) %>% 
    group_indices()
  
  sampled_groups <- sample(unique(group_ids), size)
  
  data %>% 
    filter(group_ids %in% sampled_groups)
}

# Set theme
theme_set(
  theme_classic() +
  theme(text=element_text(size=16,  family="Helvetica")))

# I use layer a lot, to rotate long x-axis labels
Rotate_x_labels <- theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))


PSI <- read_tsv("../code/kaessman_AS_dat/All.ortho.exons.PSI.tsv.gz")


RPKM <- Sys.glob("../code/kaessman_AS_dat/FromWebApp/*/RPKM.gz") %>%
  setNames(str_replace(., "../code/kaessman_AS_dat/FromWebApp/(.+?)/RPKM.gz", "\\1")) %>%
  lapply(fread, sep=' ')

dim(PSI)

RPKM.sampleList <- lapply(RPKM, colnames) %>%
  lapply(data.frame) %>%
  bind_rows(.id="Species") %>%
  dplyr::rename("sample"=2) %>%
  filter(!sample=="Names") %>%
  mutate(Species = str_to_title(Species)) %>%
  mutate(sample = paste(Species, sample, sep = "."))

ManualOrdinalStages <- read_tsv("../output/CordosoMoreira_CalibratedStageTable.tsv")

ManualOrdinalStages %>%
  mutate(Species = factor(Species, levels=c("Human", "Macaque", "Mouse", "Rat", "Rabbit", "Opossum", "Chicken"))) %>%
  ggplot(aes(y=OrdinalStage.Manual, x="")) +
  geom_text(aes(label=StageName, color=RoughlyCalibratedStage), size=2) +
  scale_y_reverse() +
  facet_wrap(~Species, nrow = 1) +
  labs(x=NULL)

OrthologousSegmentList <- read_csv("../code/kaessman_AS_dat/Supplementary_Data/Supplementary_Data_8.csv")

OrthologousSegmentList.long <- OrthologousSegmentList %>%
  mutate(human.seg.id = human) %>%
  filter(human %in% PSI$human.seg.id) %>%
  pivot_longer(names_to = "Species", values_to = "segid", -human.seg.id) %>%
  separate(segid, into=c("AS_segment", "chrom", "strand", "start", "stop"), sep=":", convert=T)

AS_segment_classifications <- Sys.glob("../code/kaessmanAnalysis/leaf2_to_AS_segments/*.collapsed.tsv.gz") %>%
  setNames(str_replace(., "../code/kaessmanAnalysis/leaf2_to_AS_segments/(.+?).collapsed.tsv.gz", "\\1")) %>%
  lapply(fread) %>%
  bind_rows(.id="Species") %>% pivot_wider(names_from = "isoform", values_from = c("PercentCoding", "n"), names_sep=".") %>%
  mutate(ChangeInCoding_InclusionToExclusion = PercentCoding.AS_seg_included - PercentCoding.AS_seg_excluded)
  

OrthologousSegmentsAndEffects <- AS_segment_classifications %>%
  dplyr::select(-Species) %>%
  inner_join(OrthologousSegmentList.long)

devAS.events.all <- read_csv("../code/kaessman_AS_dat/Supplementary_Data/Supplementary_Data_9.csv") %>%
  rename("segment.name"="...1") %>%
  pivot_longer(cols=-c(1:8), names_to = c(".value", "tissue"), values_to = c("value"), names_pattern=c("(^.+)\\.(.+$)")) %>%
  filter(pattern %in% c("d", "u")) %>%
  inner_join(AS_segment_classifications, by=c("seg.id"="AS_segment")) %>%
  mutate(tissue = str_to_title(tissue)) %>%
  dplyr::select(-ens_id)

```
### Number devAS events by species, with symmetry

```{r}
Dev.AS.Events <- read_csv("../code/kaessman_AS_dat/Supplementary_Data/Supplementary_Data_9.csv") %>%
  rename("segment.name"="...1") %>%
  pivot_longer(cols=-c(1:8), names_to = c(".value", "tissue"), values_to = c("value"), names_pattern=c("(^.+)\\.(.+$)")) %>%
  filter(!pattern %in% c("n", "-")) %>%
  mutate(start = start - 1) %>%
  mutate(len = stop - start) %>%
  mutate(LenMod3 = len %% 3) %>%
  mutate(IsSymmetric = LenMod3==0) %>%
  inner_join(OrthologousSegmentList.long %>%
               dplyr::select(-start, -stop, -strand), by=c("seg.id"="AS_segment"))


Dev.AS.Events.ToPlot <- Dev.AS.Events %>%
  filter(Species == "human") %>%
  inner_join(
    Dev.AS.Events %>%
      filter(!Species == "human"),
    by=c("human.seg.id", "tissue")
  ) %>%
  sample_frac(1) %>%
  filter(pattern.x %in% c("u", "d")) %>%
  filter(pattern.x == pattern.y) %>%
  filter(as.type.x == "CE") %>%
  add_count(tissue, human.seg.id) %>%
  filter(n>1) %>%
  distinct(human.seg.id, Species.y, seg.id.x, seg.id.y, .keep_all=T) %>%
  mutate(SymmetryConservation = case_when(
    IsSymmetric.y == IsSymmetric.x & IsSymmetric.y == T ~ "Symmetric in both",
    IsSymmetric.y == IsSymmetric.x & IsSymmetric.y == F ~ "Asymmetric in both",
    TRUE ~ "Symmetry status flips"
  ))

Dev.AS.Events.ToPlot %>%
  count(Species.y, SymmetryConservation) %>%
  ggplot(aes(x=Species.y, y=n, fill=SymmetryConservation)) +
  geom_col(position='stack') +
  Rotate_x_labels
  
```


beta beta scatters

```{r}


PSI.all <- Sys.glob("../code/kaessman_AS_dat/FromWebApp/*/PSI.gz") %>%
  setNames(str_replace(., "../code/kaessman_AS_dat/FromWebApp/(.+?)/PSI.gz", "\\1")) %>%
  lapply(fread, sep=',') %>%
  lapply(function(x) filter(x, V1 %in% devAS.events.all$segment.name)) %>%
  lapply(function(x) pivot_longer(x, names_to = "sample",values_to = "PSI",-V1)) %>%
  bind_rows() %>%
  dplyr::rename("seg.id"="V1")

RPKM.all <- Sys.glob("../code/kaessman_AS_dat/FromWebApp/*/RPKM.gz") %>%
  setNames(str_replace(., "../code/kaessman_AS_dat/FromWebApp/(.+?)/RPKM.gz", "\\1")) %>%
  lapply(fread, sep=' ') %>%
  lapply(function(x) filter(x, Names %in% devAS.events.all$Gene_name)) %>%
  lapply(function(x) pivot_longer(x, names_to = "sample",values_to = "RPKM",-Names)) %>%
  bind_rows(.id="Species") %>%
  mutate(sample = paste(str_to_title(Species), sample, sep="."))

Joined.PSI.RPKM.all <- devAS.events.all %>%
  inner_join(PSI.all) %>%
  inner_join(RPKM.all %>%
               dplyr::select(Gene_name=Names, sample, RPKM)) %>%
  separate(sample, into=c("Species", "Tissue", "StageName", "StageOrdinal"), convert=T, remove=F) %>%
  filter(Tissue == tissue) %>%
  inner_join(ManualOrdinalStages)



dat.toPlot <- Joined.PSI.RPKM.all %>%
  group_by(segment.name, tissue, Species) %>%
  summarise(cor.PSI = cor(PSI, OrdinalStage.Manual, method = "sp"),
            cor.RPKM = cor(RPKM, OrdinalStage.Manual, method='sp')) %>%
  ungroup() %>%
  inner_join(
    Joined.PSI.RPKM.all %>%
      distinct(segment.name, tissue, .keep_all=T)
  ) %>%
  mutate(ChangeInCoding_ExclusionToInclusion.Rounded = round(ChangeInCoding_InclusionToExclusion*-1)) %>%
  filter(!is.na(ChangeInCoding_ExclusionToInclusion.Rounded)) %>%
  mutate(ChangeInCoding_ExclusionToInclusion.Rounded = recode(ChangeInCoding_ExclusionToInclusion.Rounded, `1`="NMD-S", `0`="NoChange", `-1`="NMD-R"))

dat.toPlot %>%
  ggplot(aes(x=cor.PSI, y=cor.RPKM, color=tissue)) +
  geom_point(alpha=0.1) +
  geom_smooth(method='lm', se=F) +
  facet_grid(Species~ChangeInCoding_ExclusionToInclusion.Rounded)

dat.toPlot %>%
  group_by(Species, tissue, ChangeInCoding_ExclusionToInclusion.Rounded) %>%
  summarise(spearman.of_CorPSI_vs_CorRPKM = cor(cor.PSI, cor.RPKM, method='s', use="pairwise.complete.obs"), n=n()) %>%
  ungroup() %>%
  ggplot(aes(x=Species, y=tissue, fill=spearman.of_CorPSI_vs_CorRPKM)) +
  geom_raster() +
  geom_text(aes(label=n)) +
  scale_fill_gradient2() +
  facet_wrap(~ChangeInCoding_ExclusionToInclusion.Rounded, ncol=1) +
  theme(legend.position = 'bottom')

dat.toPlot %>%
  filter(tissue == "Brain" & Species == "Human") %>%
  ggplot(aes(x=cor.PSI, y=cor.RPKM)) +
  geom_point(alpha=0.1) +
  geom_smooth(method='lm', se=F) +
  facet_grid(Species~ChangeInCoding_ExclusionToInclusion.Rounded) +
  labs(x="Splicing effect over development\n(spearman rho)", y="Expression effect over development\n(spearman rho)", title="Human brain devAS events\ncorrelate with expression as expected")

  
  
```

Example event:

```{r}
ChickenConservedEvents <- Dev.AS.Events.ToPlot %>%
  group_by(human.seg.id) %>%
  filter(any(Species.y=="chicken")) %>%
  ungroup()

PSI.devASEvents <- PSI.all %>%
  filter(seg.id %in% c(ChickenConservedEvents$seg.id.x, ChickenConservedEvents$seg.id.y)) %>%
  separate(sample, into=c("Species", "tissue", "StageName", "Number"), sep="\\.", remove=F)

set.seed(0)
PSI.devASEvents %>%
  inner_join(ManualOrdinalStages) %>%
  inner_join(OrthologousSegmentList.long, by=c("seg.id"="AS_segment")) %>%
  inner_join(
    Dev.AS.Events.ToPlot %>% dplyr::select(human.seg.id, tissue) %>%
      mutate(tissue = str_to_title(tissue))
  ) %>%
  filter(human.seg.id=="hum.25142.s12:16:+:85116170:85116199") %>%
  # sample_n_of(3, human.seg.id) %>%
  # distinct(human.seg.id)
  inner_join(Dev.AS.Events.ToPlot %>%
              mutate(tissue = str_to_title(tissue)) %>%
               dplyr::select(tissue, human.seg.id)) %>%
  ggplot(aes(x=OrdinalStage.Manual, y=PSI, color=tissue)) +
  geom_point() +
  geom_smooth() +
  facet_grid(human.seg.id~Species.y, scales="free")

library(ggnewscale)

PSI.devASEvents %>%
  inner_join(ManualOrdinalStages) %>%
  inner_join(OrthologousSegmentList.long, by=c("seg.id"="AS_segment")) %>%
  inner_join(
    Dev.AS.Events.ToPlot %>% dplyr::select(human.seg.id, tissue) %>%
      mutate(tissue = str_to_title(tissue))
  ) %>%
  filter(human.seg.id=="hum.25142.s12:16:+:85116170:85116199") %>%
  mutate(Species.y = factor(Species.y, levels=c("human", "macaque", "mouse", "rat", "rabbit", "opossum", "chicken"))) %>%
  # sample_n_of(3, human.seg.id) %>%
  # distinct(human.seg.id)
  inner_join(Dev.AS.Events.ToPlot %>%
              mutate(tissue = str_to_title(tissue)) %>%
               dplyr::select(tissue, human.seg.id)) %>%
  distinct(tissue, OrdinalStage.Manual, Species.y, Number, .keep_all=T) %>%
  ggplot(aes(x=OrdinalStage.Manual, y=PSI, color=tissue)) +
  geom_point(alpha=0.5) +
  geom_smooth(se=F) +
  scale_color_manual(values=c("Brain"="blue", "Cerebellum"="red")) +
  new_scale_color() +
  geom_vline(data = . %>%
              distinct(Species.y, RoughlyCalibratedStage, .keep_all=T) %>%
               filter(!is.na(RoughlyCalibratedStage)),
            aes(xintercept=OrdinalStage.Manual, color=RoughlyCalibratedStage),
            linetype='dashed') +
  facet_wrap(~Species.y, scales="free_x") +
  labs(x="Ordinal developemental stage")

dat.toPlot
```

```{r}
devAS.ConservedWithHuman <- devAS.events.all %>%
  mutate(Species = str_replace(Species, "^(.+?)_.+$", "\\1")) %>%
  mutate(Species = factor(Species, levels=c("Human", "Macaque", "Mouse", "Rat", "Rabbit","Opossum", "Chicken"))) %>%
  filter(Species == "Human") %>%
  mutate(ChangeInCoding_ExclusionToInclusion.Rounded = round(ChangeInCoding_InclusionToExclusion*-1)) %>%
  filter(!is.na(ChangeInCoding_ExclusionToInclusion.Rounded)) %>%
  mutate(ChangeInCoding_ExclusionToInclusion.Rounded = recode(ChangeInCoding_ExclusionToInclusion.Rounded, `1`="NMD-S", `0`="NoChange", `-1`="NMD-R")) %>%
  inner_join(
      OrthologousSegmentList.long %>%
      dplyr::select(human.seg.id, seg.id = AS_segment)
      ) %>%
  inner_join(
    devAS.events.all %>%
      mutate(Species = str_replace(Species, "^(.+?)_.+$", "\\1")) %>%
      mutate(Species = factor(Species, levels=c("Human", "Macaque", "Mouse", "Rat", "Rabbit","Opossum", "Chicken"))) %>%
      filter(!Species == "Human") %>%
      mutate(ChangeInCoding_ExclusionToInclusion.Rounded = round(ChangeInCoding_InclusionToExclusion*-1)) %>%
      filter(!is.na(ChangeInCoding_ExclusionToInclusion.Rounded)) %>%
      mutate(ChangeInCoding_ExclusionToInclusion.Rounded = recode(ChangeInCoding_ExclusionToInclusion.Rounded, `1`="NMD-S", `0`="NoChange", `-1`="NMD-R")) %>% 
      inner_join(
        OrthologousSegmentList.long %>%
        dplyr::select(human.seg.id, seg.id = AS_segment)
        ),
    by=c("tissue", "human.seg.id"), suffix=c(".human", ".other")) %>%
  filter(pattern.human == pattern.other)

Conserved.UnproductiveEvents <- devAS.ConservedWithHuman %>%
  mutate(Conserved_NMD_Result = if_else(ChangeInCoding_ExclusionToInclusion.Rounded.human == ChangeInCoding_ExclusionToInclusion.Rounded.other, "Conserved result", "Unconserved result"))  %>%
  filter(Conserved_NMD_Result == "Conserved result" & ChangeInCoding_ExclusionToInclusion.Rounded.other %in% c("NMD-R", "NMD-S"))

mart <- useEnsembl("ensembl","hsapiens_gene_ensembl")
genes <- getBM(attributes = c('ensembl_gene_id', 'hgnc_symbol'),
               mart = mart)

Conserved.UnproductiveEvents %>%
  inner_join(genes, by=c("Gene_name.human"="ensembl_gene_id")) %>%
  dplyr::select(segment.name.human, Gene_name.human, hgnc_symbol, tissue,Result=ChangeInCoding_ExclusionToInclusion.Rounded.other, Species=Species.other, AS_type=as.type.human,pattern=pattern.other ) %>%
  mutate(Species.Tissue = paste(Species, tissue)) %>%
  arrange(Species, tissue) %>%
  group_by(segment.name.human, Gene_name.human,  hgnc_symbol,Result, AS_type, pattern ) %>%
  summarise(Species.Tissues = paste0(Species.Tissue,collapse = ', '), NumberConservedSpeciesTissuePairs=n()) %>%
  ungroup() %>%
  arrange(desc(NumberConservedSpeciesTissuePairs), hgnc_symbol) %>%
  dplyr::select(hgnc_symbol, Result, NumberConservedSpeciesTissuePairs, Species.Tissues) %>%
  knitr::kable()


```

